### PR TITLE
[Bugs] Fix the bugs list of sync job

### LINKED ISSUE #6704: [Bug] Bug list of sync job (Mysql synchronization)
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master branch

### What's Wrong?

QA and I found a lot of bugs during sync job's functional testing which lists as follows:
```
1、The sync jobs are not cancelled after deleting the database.
2、The MySQL and Doris tables should have a one-to-one correspondence. 
      If they are not, they should fail when creating the task.
3、When the cluster has multiple FE, the non-master will core when replay create the 
      sync job.
4、Inconsistent data when updating key column
5、Failed to synchronize data when there are multiple tables in single sync job.
6、After restarting the master, resuming the paused sync job will fail.
```

### What You Expected?

List the bugs and fix them.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xy720**
## Proposed changes

QA and I found a lot of bugs during sync job's functional testing. This pr will fix them as follows:

```
1、Fix bug that the sync jobs are not cancelled after deleting the database.
2、The MySQL and Doris tables should have a one-to-one correspondence. 
      If they are not, they should fail when creating the task.
3、When the cluster has multiple FE, the non-master will core when replay create the sync job.
4、Inconsistent data when updating key column
5、Failed to synchronize data when there are multiple tables in single sync job.
6、After restarting the master, resuming the paused syncjob will fail.
```

related to #6287 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6704 ) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJob.java [hunk 1]
```diff
@@ -152,6 +186,65 @@ public synchronized void updateState(JobState newState, boolean isReplay) {
         }
     }
 
+    private void checkStateTransform(JobState newState) throws UserException {
+        CHECK: {
+            switch (jobState) {
+                case RUNNING:
+                    if (newState == JobState.RUNNING) {
+                        break CHECK;
```
Comment (pengxiangyu) [inline fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJob.java:194]: 这里直接throw比较好，这样不需要外面的CHECK，结尾处也不需要再return了。逻辑比较清晰

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJob.java [hunk 2]
```diff
@@ -152,6 +186,65 @@ public synchronized void updateState(JobState newState, boolean isReplay) {
...
+                    }
+                    break;
+                case PAUSED:
+                    if (newState == JobState.PAUSED || newState == JobState.RUNNING) {
+                        break CHECK;
```
Comment (pengxiangyu) [inline fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJob.java:199]: 同上，这一段整体改造一下吧。

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJobManager.java [hunk 3]
```diff
@@ -281,9 +296,6 @@ public void readField(DataInput in) throws IOException {
         int size = in.readInt();
         for (int i = 0; i < size; i++) {
             SyncJob syncJob = SyncJob.read(in);
-            if (!syncJob.isCompleted()) {
-                syncJob.updateState(JobState.PENDING, true);
-            }
             unprotectedAddSyncJob(syncJob);
```
Comment (pengxiangyu) [inline fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJobManager.java:299]: 这里不用加锁么？调用unprotectedAddSyncJob的时候没有检查同名jobName是否已存在，应该要做下检查吧，否则同名任务会被直接覆盖。

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncLifeCycle.java [hunk 4]
```diff
@@ -55,10 +55,6 @@ public void run() {
     }
 
     public void stop() {
-        if (!isStart()) {
-            throw new RuntimeException(this.getClass().getName() + " isn't start , please check");
```
Comment (pengxiangyu) [inline fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncLifeCycle.java:59]: 这块外面没有catch这个异常的地方，没什么意义。如果isStart()为false也存在重复stop的情况，建议直接按stop成功来处理。

**Review by pengxiangyu — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJob.java (hunk 1 above)
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJob.java]: ok，改成直接throw了

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJobManager.java (hunk 3 above)
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncJobManager.java:299]: readfield方法理论上只会由checkpoint线程来调用，写入的catalog和现在FE运行时的catalog不是一个对象，所以这里不会有线程安全的问题。我们只需要在创建syncjob的时候检查同名，就可以保证checkpoint时不会出现重复名称了，这一检查逻辑在DdlExecutor里。

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncLifeCycle.java (hunk 4 above)
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncLifeCycle.java:59]: 没错，这一块我去掉抛出异常，就是想重复的stop都按成功处理。

Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/load/sync/SyncLifeCycle.java:59]: 改成了如果重复stop，后面的逻辑都会跳过

**Review by pengxiangyu — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**